ui/ducks/send/swap-and-send-utils.ts

```diff
@@ -208,3 +210,50 @@ export async function getSwapAndSendQuotes(request: Request): Promise<Quote[]> {
 
   return newQuotes;
 }
+
+const CHAIN_ID_TO_API_NAME: Record<string, string> = {
+  [CHAIN_IDS.MAINNET]: 'ethereum',
+  [CHAIN_IDS.BSC]: 'bsc',
+  [CHAIN_IDS.POLYGON]: 'polygon',
+  [CHAIN_IDS.AVALANCHE]: 'avalanche',
+  [CHAIN_IDS.ARBITRUM]: 'arbitrum',
+  [CHAIN_IDS.OPTIMISM]: 'optimism',
+  [CHAIN_IDS.ZKSYNC_ERA]: 'zksync',
+  [CHAIN_IDS.LINEA_MAINNET]: 'linea',
+  [CHAIN_IDS.BASE]: 'base',
+};
+
+export async function getDisabledSwapAndSendNetworksFromAPI(): Promise<
+  string[]
+> {
+  try {
+    const blockedChains: string[] = [];
+
+    const url = `${BASE_URL}/featureFlags`;
+
+    const featureFlagResponse = await fetchWithCache({
+      url,
+      fetchOptions: {
+        method: 'GET',
+        headers: { 'X-Client-Id': SWAPS_CLIENT_ID },
+      },
+      cacheOptions: { cacheRefreshTime: 0, timeout: MINUTE * 5 },
+      functionName: 'getDisabledSwapAndSendNetworks',
+    });
```

Comment:
Actually, would it be possible to reuse `state.metamask.swapsState.swapsFeatureFlags` background state here? I'm thinking the fetch call can be added to `useUpdateSwapsState`

Reply:
Would prefer not to since it decouples the logic (i.e., we have to reference the swap state, and rely on said state getting updated through another couple) and prevents us from just getting the flags whenever the send flow is initialized

Reply:
aren't we already doing that in the hook though?

Reply:
Yeah, I'm not a fan of the pattern though honestly. It was just the easiest way to ensure the swap state is ready (I just copied a bunch of stuff over). I worry that going through the swaps state too much will make our logic more likely to break when something in swaps is changed – we got burnt when a selector for non-1559 gas was refactored during #23703

Reply:
Fair point, lgtm. Although this may become a more common pattern as we introduce more combined experiences, like Swap+Bridge and Swap+Send

> going through the swaps state too much will make our logic more likely to break when something in swaps is changed

Fwiw I think this could be framed as a benefit as well - if there's a breaking swaps change it would be more ideal to break Swap+Send at PR or QA time, rather than failing silently when we start getting unexpected data

Reply:
All valid points, but without TypeScript things could easily slip through the gaps.